File: RedSheet.DbEntities/Models/OurRequirementDetail.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("OurRequirementDetails", Schema = "dbo")]
    [Serializable]
    public partial class OurRequirementDetail
    {

        [MaxLength(150)]
        public string FourStep { get; set; }

        [Required]
        [MaxLength(150)]
        public string ldo { get; set; }

        [Required]
        [MaxLength(150)]
        public string mdo { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int OurRequirementDetailId { get; set; }

        [Required]
        [MaxLength(150)]
        public string Requirement { get; set; }

        [MaxLength(150)]
        public string SecondStep { get; set; }

        [MaxLength(150)]
        public string ThirdStep { get; set; }

        [RelationshipTableAttribue("ApplicationObjects", "dbo")]
        public PainFactor PainFactorId { get; set; }

        [RelationshipTableAttribue("ProjectRequirements", "dbo")]
        //Relationships
        public int ProjectRequirementId { get; set; }

        [ForeignKey("ProjectRequirementId")]
        public ProjectRequirement ProjectRequirement { get; set; }
        [InverseProperty("OurRequirementDetail")]
        public ICollection<ProjectZoma> ProjectZomas { get; set; }


        public OurRequirementDetail()
        {
            this.ProjectZomas = new HashSet<ProjectZoma>();

        }

    }
}

File: redsheet-portal/src/app/database-models/our-requirement-detail.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectRequirement, ProjectZoma,  vOurRequirementDetailRecord  } from './'
export class OurRequirementDetail {
    constructor(ourRequirementDetail?: OurRequirementDetail  | vOurRequirementDetailRecord )  {
        let properties = [ "fourStep", "ldo", "mdo", "ourRequirementDetailId", "requirement", "secondStep", "thirdStep", "painFactorId", "projectRequirementId", "projectZomas", "projectModuleId",];
        for (let property of properties)
            if (ourRequirementDetail && ourRequirementDetail[property])
                this[property] = ourRequirementDetail[property];
    }

    @maxLength(150)
	fourStep : string =   undefined;

    @required()
    @maxLength(150)
	ldo : string =   undefined;

    @required()
    @maxLength(150)
	mdo : string =   undefined;

	ourRequirementDetailId : number =   0 ;

    @required()
    @maxLength(150)
	requirement : string =   undefined;

    @maxLength(150)
	secondStep : string =   undefined;

    @maxLength(150)
	thirdStep : string =   undefined;

    @range(0,2147483647)
	painFactorId : number =   undefined;

    @range(0,2147483647)
	projectRequirementId : number =   undefined;
	projectRequirement : ProjectRequirement  ;
	@nested(ProjectZoma)
	projectZomas: ProjectZoma[];


	projectModuleId : number =   undefined;

}
